mesonbuild/backend/backends.py

```diff
@@ -2080,3 +2079,25 @@ def compile_target_to_generator(self, target: build.CompileTarget) -> build.Gene
         all_sources = T.cast('_ALL_SOURCES_TYPE', target.sources) + T.cast('_ALL_SOURCES_TYPE', target.generated)
         return self.compiler_to_generator(target, target.compiler, all_sources,
                                           target.output_templ, target.depends)
+
+    def is_unity(self, target: build.BuildTarget) -> bool:
+        if isinstance(target, build.CompileTarget):
+            return False
+        val = self.get_target_option(target, 'unity')
+        if val == 'on':
+            return True
+        if val == 'off':
+            return False
+        if val == 'subprojects':
+            return target.subproject != ''
```

Comment:
The change I have consistently gone to was to separate "data storage" from "evaluation code". Before this the logic of how option values were evaluated was all over the place. Not to mention that some of the data transformations were silently lossy. This made it hard to answer questions like "where _exactly_ does the value of option X for Y come from".

Because of this all data containers like build targets _only_ store the original data they were given as close to the original form as possible. They do not have logic. They do not do transformations. All that is computed in one place and if you ever need to debug issues with it you know exactly the one place where you should set your debugger breakpoint.

This is a bit unfortunate for Unity since it has the "for subprojects" value. If we had had this functionality already it would never have been needed. But it's there now, so we have to deal with it and eventually deprecate and remove the "for subprojects only" value from it.